### PR TITLE
Create Benchmarking suite

### PR DESCRIPTION
**Pull request by squat**
This PR creates the benchmarking test script. This script creates
the benchmarking resources on a cluster, runs the benchmarking binary,
and collects test results.

To run the test:
1. create a cluster
2. `export KUBECONFIG=...`
3. `make test-benchmark`

The results of this benchmarking test can be analyzed using the
newly added test/plot.py script.

To generate results: `make benchmark.pdf`

Finally, test/timeseries.txt contains example timeseries used by the
benchmark binary to create payloads to send to the Telemeter server
against which the benchmark tests are run.

To regenerate the timeseries list:
1. create a cluster
2. `export KUBECONFIG=...`
3. `make test/timeseries.txt`

### CODE REVIEW

**Comment by jeremyeder** [comment]
@akrzos @chaitanyaenr

File: test/plot.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
+import matplotlib.pyplot as plt
```
Comment (s-urbaniak) [inline test/plot.py]: please add a `#!/usr/bin/env python3` to be non ambiguous

File: Makefile [hunk 2]
```diff
@@ -38,19 +40,37 @@ check: lint
...
+	$(JSONNET_BIN) jsonnet/benchmark.jsonnet -J jsonnet/vendor -m manifests/benchmark
 	$(JSONNET_BIN) jsonnet/client.jsonnet -J jsonnet/vendor -m manifests/client
 	$(JSONNET_BIN) jsonnet/server.jsonnet -J jsonnet/vendor -m manifests/server
 	$(JSONNET_BIN) jsonnet/prometheus.jsonnet -J jsonnet/vendor -m manifests/prometheus
 	@for f in $$(find manifests -type f); do\
 	    cat $$f | $(GOJSONTOYAML_BIN) > $$f.yaml && rm $$f;\
 	done
 
+benchmark.pdf: $(BENCHMARK_RESULTS)
+	find ./benchmark -type f -name '*.json' -print0 | xargs -l -0 python test/plot.py && pdfunite benchmark/*.pdf $@
```
Comment (s-urbaniak) [inline Makefile]: same here, please use `python3` explicitely.

Comment (s-urbaniak) [reply Makefile]: please document the dependencies to python3 and poppler.

**Comment by s-urbaniak** [comment]
this looks great! :tada: There is a lot of script/Makefile/Python goodness in here :+1: 
Let's just add some readme, a copy&paste from the PR description should be enough and the prerequisites (python, poppler).

**Comment by metalmatze** [comment]
Wow, this is amazing! :100: :+1:

**Pull request merged**